contracts/zero-ex/contracts/src/transformers/bridges/mixins/MixinCompound.sol

```diff
@@ -0,0 +1,109 @@
+    /// @return status code of whether the mint was successful or not
+    function mint(uint mintAmountInUnderlying) external returns (uint);
+    /// @dev redeems specified amount of cTokens and returns the underlying token to the sender
+    /// @param redeemTokensInCtokens amount of cTokens to redeem for underlying collateral
+    /// @return status code of whether the redemption was successful or not
+    function redeem(uint redeemTokensInCtokens) external returns (uint);
+}
+/// @dev Minimal CEther interface
+interface ICEther {
+    /// @dev deposits the amount of Ether sent as value and return mints cEther for the sender
+    function mint() payable external;
+    /// @dev redeems specified amount of cETH and returns the underlying ether to the sender
+    /// @dev redeemTokensInCEther amount of cETH to redeem for underlying ether
+    /// @return status code of whether the redemption was successful or not
+    function redeem(uint redeemTokensInCEther) external returns (uint);
+}
+
+contract MixinCompound {
+    using LibERC20TokenV06 for IERC20TokenV06;
+    using LibSafeMathV06 for uint256;
+
+    IEtherTokenV06 private immutable WETH;
+
+    constructor(IEtherTokenV06 weth)
+        public
+    {
+        WETH = weth;
+    }
+
+    uint256 constant private COMPOUND_SUCCESS_CODE = 0;
+
+    function _tradeCompound(
+        IERC20TokenV06 sellToken,
+        IERC20TokenV06 buyToken,
+        uint256 sellAmount,
+        bytes memory bridgeData
+    )
+        internal
+        returns (uint256)
+    {
+        (address cTokenAddress) = abi.decode(bridgeData, (address));
+        uint256 beforeBalance = buyToken.balanceOf(address(this));
+
+
+        if (address(buyToken) == cTokenAddress) {
+            if (address(sellToken) == address(WETH)) {
+                // ETH/WETH -> cETH
+                ICEther cETH = ICEther(cTokenAddress);
+                // Compound expects ETH to be sent with mint call
+                WETH.withdraw(sellAmount);
+                // NOTE: cETH mint will revert on failure instead of returning a status code
+                cETH.mint{value: sellAmount}();
+            } else {
+                sellToken.approveIfBelow(
+                    cTokenAddress,
+                    sellAmount
+                );
+                // Token -> cToken
+                ICToken cToken = ICToken(cTokenAddress);
+                require(cToken.mint(sellAmount) == COMPOUND_SUCCESS_CODE, "MixinCompound/FAILED_TO_MINT_CTOKEN");
+            }
+        } else if (address(sellToken) == address(cTokenAddress)) {
+            if (address(buyToken) == address(WETH)) {
+                // cETH -> ETH/WETH
+                ICEther cETH = ICEther(cTokenAddress);
+                require(cETH.redeem(sellAmount) == COMPOUND_SUCCESS_CODE, "MixinCompound/FAILED_TO_REDEEM_CETHER");
+                uint256 receivedEtherBalance = address(this).balance;
+                WETH.deposit{value: receivedEtherBalance}();
```

Comment:
I think for correctness we want to do a balance delta here instead of `this.balance`. The FlashWallet could theoretically hold ETH for protocol fees, or maybe we'll have another use case for it in the future, and this would gobble it up.